protobuf/timestamp.go

```diff
@@ -0,0 +1,36 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package protobuf
+
+import (
+	"time"
+
+	"github.com/gogo/protobuf/types"
+)
+
+// ToTimestamp creates protobuf's Timestamp from time.Time
+func ToTimestamp(from time.Time) *types.Timestamp {
+	return &types.Timestamp{
+		Seconds: int64(from.Second()),
+		Nanos:   int32(from.Nanosecond()),
+	}
+}
+
+// FromTimestamp creates time.Time from protobuf's Timestamp
+func FromTimestamp(from *types.Timestamp) time.Time {
```

Comment:
Note that protobuf/types already implements this: https://github.com/gogo/protobuf/blob/b03c65ea87cdc3521ede29f62fe3ce239267c1bc/types/timestamp.go#L88-L98

There's some validation there (which I didn't dig into yet), but the other case is a nil timestamp.

Reply:
Oh, how didn't I notice that. Let me fix.